game-core/src/main/java/games/strategy/triplea/delegate/UndoablePlacement.java

```diff
@@ -57,10 +57,13 @@ protected final void undoSpecific(final IDelegateBridge bridge) {
 
   @Override
   public final String getMoveLabel() {
-    if (m_producerTerritory != m_placeTerritory) {
-      return m_producerTerritory.getName() + " -> " + m_placeTerritory.getName();
-    }
-    return m_placeTerritory.getName();
+    return getMoveLabel(" -> ");
+  }
+
+  private String getMoveLabel(final String separator) {
+    return m_producerTerritory.equals(m_placeTerritory)
+        ? m_placeTerritory.getName()
+        : m_producerTerritory.getName() + separator + m_placeTerritory.getName();
```

Comment:
Interesting, I'd have used
```java
return (m_producerTerritory.equals(m_placeTerritory)
? ""
: m_producerTerritory.getName() + separator) + m_placeTerritory.getName();
```

or similar, as that makes it more clear to me that the place territory is only being added with the seperator if it's different than the producer territory.

But this is fine as well.